src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs

```diff
@@ -0,0 +1,56 @@
+// <copyright file="PeriodicExportingMetricReader.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OpenTelemetry.Metrics
+{
+    public class PeriodicExportingMetricReader : MetricReader
+    {
+        private MetricExporter exporter;
+        private Task exportTask;
+        private CancellationTokenSource token;
+
+        public PeriodicExportingMetricReader(MetricExporter exporter, int exportIntervalMs)
+        {
+            this.exporter = exporter;
+            this.token = new CancellationTokenSource();
+            this.exportTask = new Task(() =>
```

Comment:
Probably add a TODO here, I hope we can make this running on a dedicated thread (so we can use thread runaway time to see the actual serialization/exporting CPU consumption, and CSwitch). The idea is very similar to how Finalizers run in their dedicated threads.